File: src/domain/models/kikurage-state.ts
import {assertIsDefined} from '../../lib/assert';

export interface Model {
  readonly temperature: number;
  readonly humidity: number;
  readonly message: string;
  readonly typeString: string;
  readonly advice: string;
}

export function factory(kikurageState: Model): Model {
  assertIsDefined(kikurageState.temperature);

  return {
    temperature: kikurageState.temperature,
    humidity: kikurageState.humidity,
    message: kikurageState.message,
    typeString: kikurageState.typeString,
    advice: kikurageState.advice,
  };
}

File: native/app/src/main/java/com/shusuke/kikurage/entity/response/KikurageState.kt
package com.shusuke.kikurage.entity.response

import com.google.firebase.firestore.PropertyName

// NOTE: To avoid decode exception, have to set up default values

data class KikurageState (
    @PropertyName("temperature") val temperature: Int = 0,
    @PropertyName("humidity") val humidity: Int = 0,
    @PropertyName("message") val message: String = "",
    @PropertyName("judge") val typeString: String = "",
    @PropertyName("advice") val advice: String = ""
)